### PR TITLE
Add file `.php-version` support

### LINKED ISSUE #33: [PHP] Allow overriding PHP's binary based on a local `.php-version` file
The file should only contain a version flag.

This version flag must be used to select a PHP version that is visible in the `rymfony php:list` command.

The list of PHP binaries is also available in  the `$HOME/.rymfony/php-versions.json` file (if the `php:list` command was called at least once).

Also, a version flag like `7.2` should match the **latest** 7.2 version available in all binaries. If you for instance have 7.2.1 and 7.2.5, the latest should be selected.

### PR DESCRIPTION
**Pull request by macintoshplus**
close #33

### CODE REVIEW

File: src/php/binaries.rs [hunk 1]
```diff
@@ -26,6 +27,54 @@ pub(crate) fn current() -> String {
     "php".to_string()
 }
 
+
```
Comment (Pierstoval) [inline src/php/binaries.rs:30]: Unnecessary extra line
```suggestion
```

File: src/php/binaries.rs [hunk 2]
```diff
@@ -26,6 +27,54 @@ pub(crate) fn current() -> String {
...
+    let project_php_version_file_path = env::current_dir().unwrap().join(".php-version");
+
+    let mut php_version = if project_php_version_file_path.exists() {
+        read_to_string(project_php_version_file_path).unwrap()
+    } else { String::from("") };
+
+    php_version = php_version.trim().to_string();
+
+    if php_version != "" {
+        trace!(".php-version defined {}", php_version);
```
Comment (Pierstoval) [inline src/php/binaries.rs:43]:
```suggestion
        debug!("PHP version set to {} from \".php-version\" file.", php_version);
```

File: src/php/binaries.rs [hunk 3]
```diff
@@ -26,6 +27,54 @@ pub(crate) fn current() -> String {
...
+    }
+
+    let mut system = String::from("");
+    let mut user_selected = String::from("");
+    let mut user_selected_version = String::from("");
+
+    for (_version, _binary) in _binaries {
+        trace!("Work for {}, current php selected version {}", _version.version(), user_selected_version);
```
Comment (Pierstoval) [inline src/php/binaries.rs:51]: I don't see why we should trace this information 🤔 

```suggestion
```

File: src/php/binaries.rs [hunk 4]
```diff
@@ -26,6 +27,54 @@ pub(crate) fn current() -> String {
...
+
+        if php_version != "" && _version.version().starts_with(&php_version) {
+            trace!("Version matching");
```
Comment (Pierstoval) [inline src/php/binaries.rs:54]: This could be removed too IMO
```suggestion
```

File: src/php/binaries.rs [hunk 5]
```diff
@@ -26,6 +27,54 @@ pub(crate) fn current() -> String {
...
+            if user_selected_version.eq("") || user_selected_version.as_str() < _version.version() {
+                user_selected_version = String::from(_version.version());
+                user_selected = _binary.preferred_sapi().to_string();
+                trace!("New PHP version selected {}", user_selected_version);
```
Comment (Pierstoval) [inline src/php/binaries.rs:58]:
```suggestion
```

File: src/php/binaries.rs [hunk 6]
```diff
@@ -85,7 +134,7 @@ fn binaries_from_dir(path: PathBuf) -> HashMap<PhpVersion, PhpBinary> {
         // Canonicalize on Windows leaves the "\\?" prefix on canonicalized paths.
         // Let's not use it, they should be absolute anyway on Windows, so they're usable.
         #[cfg(not(target_family = "windows"))]
-        let binary: PathBuf = binary.canonicalize().unwrap();
+            let binary: PathBuf = binary.canonicalize().unwrap();
```
Comment (Pierstoval) [inline src/php/binaries.rs:137]:
```suggestion
        let binary: PathBuf = binary.canonicalize().unwrap();
```

File: src/php/binaries.rs [hunk 7]
```diff
@@ -26,6 +27,54 @@ pub(crate) fn current() -> String {
     "php".to_string()
 }
```
Comment (Pierstoval) [inline src/php/binaries.rs:28]: The `current()` function is now unused, isn't it? Maybe it could be removed :)

**Review by Pierstoval — CHANGES_REQUESTED** [review]
Small requests, I don't think `trace`-ing everything is really relevant now.
It's really useful when debugging, but not everyone will actually need this :)

For the rest, looks goot to me 👍

File: src/php/binaries.rs (hunk 7 above)
Comment (macintoshplus) [reply src/php/binaries.rs]: Yes, deleted

**Comment by Pierstoval** [comment]
Thanks @macintoshplus

**Pull request merged**